src/type/matrix/SparseMatrix.js

```diff
@@ -709,34 +709,38 @@ export const createSparseMatrixClass = /* #__PURE__ */ factory(name, dependencie
    *       resize().
    *
    * @memberof SparseMatrix
-   * @param {number[]} size           The new size the matrix should have.
+   * @param {number[]} sizes          The new size the matrix should have.
    * @param {boolean} [copy]          Return a reshaped copy of the matrix
    *
    * @return {Matrix}                 The reshaped matrix
    */
-  SparseMatrix.prototype.reshape = function (size, copy) {
+  SparseMatrix.prototype.reshape = function (sizes, copy) {
     // validate arguments
-    if (!isArray(size)) { throw new TypeError('Array expected') }
-    if (size.length !== 2) { throw new Error('Sparse matrices can only be reshaped in two dimensions') }
+    if (!isArray(sizes)) { throw new TypeError('Array expected') }
+    if (sizes.length !== 2) { throw new Error('Sparse matrices can only be reshaped in two dimensions') }
 
     // check sizes
-    size.forEach(function (value) {
-      if (!isNumber(value) || !isInteger(value) || value < 0) {
-        throw new TypeError('Invalid size, must contain positive integers ' +
-                            '(size: ' + format(size) + ')')
+    sizes.forEach(function (value) {
+      if (!isNumber(value) || !isInteger(value) || value <= -2 || value === 0) {
+        throw new TypeError('Invalid size, must contain positive integers or -1 ' +
+          '(size: ' + format(sizes) + ')')
       }
     })
 
+    const currentLength = this._size[0] * this._size[1]
+    sizes = processSizesWildcard(sizes, currentLength)
```

Comment:
I kept the variable name. Reading `newSizes` or `processedSizes` was just akward